#! /bin/sh /usr/share/dpatch/dpatch-run
## 09_amd64_epg.dpatch by Thomas Schmidt <tschmidt@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Fixes the crc function to work under amd64 too 
## DP: (closes: #300702)

@DPATCH@
diff -urNad vdr-1.2.6/libdtv/libsi/si_parser.c /tmp/dpep.Y5cRRE/vdr-1.2.6/libdtv/libsi/si_parser.c
--- vdr-1.2.6/libdtv/libsi/si_parser.c	2003-08-15 15:56:02.000000000 +0200
+++ /tmp/dpep.Y5cRRE/vdr-1.2.6/libdtv/libsi/si_parser.c	2005-03-21 18:13:17.285886176 +0100
@@ -1278,7 +1278,7 @@
 
 // CRC32 lookup table for polynomial 0x04c11db7
 
-static u_long crc_table[256] = {
+static __u32 crc_table[256] = {
 	0x00000000, 0x04c11db7, 0x09823b6e, 0x0d4326d9, 0x130476dc, 0x17c56b6b,
 	0x1a864db2, 0x1e475005, 0x2608edb8, 0x22c9f00f, 0x2f8ad6d6, 0x2b4bcb61,
 	0x350c9b64, 0x31cd86d3, 0x3c8ea00a, 0x384fbdbd, 0x4c11db70, 0x48d0c6c7,
@@ -1326,7 +1326,7 @@
 static u_long crc32 (char *data, int len)
 {
 	register int i;
-	u_long crc = 0xffffffff;
+	__u32 crc = 0xffffffff;
 
 	for (i=0; i<len; i++)
 		crc = (crc << 8) ^ crc_table[((crc >> 24) ^ *data++) & 0xff];
